### PR TITLE
Splice helper function should check for side-effect in the prototype.

### PR DESCRIPTION
**Pull request by akroshg**
Splice has fast path which did not check the prototype (say you have proxy) properly which resulted to incorrect result. Fixed that by using the side-effect macro to take the slower and observable code path.

### CODE REVIEW

**Comment by akroshg** [comment]
@rajatd please have a look at this one.

File: test/Array/protoLookupWithGetters.baseline [hunk 1]
```diff
@@ -1,35 +1,35 @@
 Test case 1
-d1,d2,d3,d4,p5,p6,p7,p8,p9
+p0,p1,p2,p3,p4,p5,p6,p7,p8
```
Comment (akroshg) [inline test/Array/protoLookupWithGetters.baseline]: earlier result was incorrect.

File: test/Array/protoLookupWithGetters.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ write("Test case 1");
 
 for(var i =0;i<10;i++)
 {
-  Object.defineProperty(Array.prototype, i, { get: function (i) { return function () { return "p"+i; } }(i), configurable: true, enumerable: true });
+  Object.defineProperty(Array.prototype, i, { get: function (i) { return function () { return "p"+i; } }(i), set : function(a) {}, configurable: true, enumerable: true });
```
Comment (akroshg) [inline test/Array/protoLookupWithGetters.js]: made this change as you need to have setter.

**Comment by rajatd** [comment]
LGTM

**Pull request merged**